DragaliaAPI/Controllers/Dragalia/AbilityCrestController.cs

```diff
@@ -152,4 +156,88 @@ public async Task<DragaliaResult> ResetPlusCount(AbilityCrestResetPlusCountReque
         UpdateDataList updateDataList = await this.updateDataService.SaveChangesAsync();
         return Ok(new AbilityCrestResetPlusCountData() { update_data_list = updateDataList });
     }
+
+    [Route("get_ability_crest_set_list")]
+    [HttpPost]
+    public async Task<DragaliaResult> GetAbilityCrestSetList(
+        AbilityCrestGetAbilityCrestSetListRequest request
+    )
+    {
+        List<DbAbilityCrestSet> dbAbilityCrestSets = await abilityCrestRepository.AbilityCrestSets
+            .OrderBy(x => x.AbilityCrestSetNo)
+            .ToListAsync();
+
+        int index = 0;
+
+        IEnumerable<AbilityCrestSetList> abilityCrestSetList = Enumerable
+            .Range(1, 54)
+            .Select(
+                x =>
+                    index < dbAbilityCrestSets.Count()
+                    && dbAbilityCrestSets[index].AbilityCrestSetNo == x
+                        ? dbAbilityCrestSets[index++]
+                        : new DbAbilityCrestSet(DeviceAccountId, x)
```

Comment:
This doesn't respect user impersonation, though the value of DeviceAccountId is also never used so its fine - however I'd replace it with an empty string as a precaution if we do decide to completely remove the DeviceAccountId property in the future